plugins/offboard/offboard_impl.cpp

```diff
@@ -106,6 +106,88 @@ void OffboardImpl::receive_command_result(MAVLinkCommands::Result result,
     }
 }
 
+void OffboardImpl::set_translation_ned(TranslationNED translation_ned)
+{
+    _mutex.lock();
+    _translation_ned = translation_ned;
+
+    if (_mode != Mode::TRANSLATION_NED) {
+        if (_call_every_cookie) {
+            // If we're already sending other setpoints, stop that now.
+            _parent->remove_call_every(_call_every_cookie);
+            _call_every_cookie = nullptr;
+        }
+        // We automatically send NED setpoints from now on.
+        _parent->add_call_every(
+            [this]() { send_translation_ned(); }, SEND_INTERVAL_S, &_call_every_cookie);
+
+        _mode = Mode::TRANSLATION_NED;
+    } else {
+        // We're already sending these kind of setpoints. Since the setpoint change, let's
+        // reschedule the next call, so we don't send setpoints too often.
+        _parent->reset_call_every(_call_every_cookie);
+    }
+    _mutex.unlock();
+
+    // also send it right now to reduce latency
+    send_translation_ned();
+}
+
+void OffboardImpl::set_position_ned_yaw(Offboard::PositionNEDYaw position_ned_yaw)
+{
+    TranslationNED translation_ned = {position_ned_yaw.north_m,
+                                      position_ned_yaw.east_m,
+                                      position_ned_yaw.down_m,
+                                      NAN,
+                                      NAN,
+                                      NAN,
+                                      position_ned_yaw.yaw_deg,
+                                      NAN};
+    set_translation_ned(translation_ned);
+}
+
+void OffboardImpl::set_velocity_altitude_ned_yaw(
+    Offboard::VelocityAltitudeNEDYaw velocity_altitude_ned_yaw)
+{
+    TranslationNED translation_ned = {NAN,
+                                      NAN,
+                                      velocity_altitude_ned_yaw.down_m,
+                                      velocity_altitude_ned_yaw.north_m_s,
+                                      velocity_altitude_ned_yaw.east_m_s,
+                                      NAN,
+                                      velocity_altitude_ned_yaw.yaw_deg,
+                                      NAN};
```

Comment:
The way this is written is actually hard to read because it is not clear what the fields are and what is set to `NAN`.
I think it's not self-explanatory and can lead to mistakes. What about this way?
```
TranslationNED translation_ned{NAN, NAN, NAN, NAN, NAN, NAN, NAN, NAN};
translation_ned.down_m = position_ned_yaw.down_m;
...
```